src/data/finsupp.lean

```diff
@@ -1383,6 +1383,10 @@ ext $ λ a', by by_cases a = a';
   [{ subst h, simp only [smul_apply', single_eq_same] },
    simp only [h, smul_apply', ne.def, not_false_iff, single_eq_of_ne, smul_zero]]
 
+@[simp] lemma smul_single_self {R : semiring γ}
+  (c : γ) (a : α) (b : γ) : c • finsupp.single a b = finsupp.single a (c * b) :=
+smul_single _ _ _
+
```

Comment:
There was a linting error complaining correctly that `monoid_algebra.smul_single` and `add_monoid_algebra.smul_single` were redundant. I replaced them both with this lemma. I don't much like the name, however, so feel free to change it.

Reply:
I think I've missed the fact that we don't have `monoid_algebra.single`, only `finsupp.single`.

Reply:
BTW, why this is not redundant? Why lean doesn't apply `smul_single` then `smul_eq_mul`?

Reply:
It is. Removed.